owncloudData/src/main/java/com/owncloud/android/data/webfinger/datasources/implementation/OCRemoteWebFingerDatasource.kt

```diff
@@ -0,0 +1,73 @@
+/**
+ * ownCloud Android client application
+ *
+ * Copyright (C) 2022 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.data.webfinger.datasources.implementation
+
+import com.owncloud.android.data.ClientManager
+import com.owncloud.android.data.executeRemoteOperation
+import com.owncloud.android.data.webfinger.datasources.RemoteWebFingerDatasource
+import com.owncloud.android.domain.webfinger.model.WebFingerRel
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.authentication.OwnCloudCredentialsFactory
+import com.owncloud.android.lib.resources.webfinger.services.WebFingerService
+
+class OCRemoteWebFingerDatasource(
+    private val webfingerService: WebFingerService,
+    private val clientManager: ClientManager,
+) : RemoteWebFingerDatasource {
+
+    override fun getInstancesFromWebFinger(
+        lookupServer: String,
+        rel: WebFingerRel,
+        username: String
+    ): List<String> {
+        val ownCloudClient = clientManager.getClientForAnonymousCredentials(lookupServer, false)
+
+        return executeRemoteOperation {
+            webfingerService.getInstancesFromWebfinger(
```

Comment:
In the library as well 😀 
```suggestion
            webfingerService.getInstancesFromWebFinger(
```